provider/spark.go

```diff
@@ -1129,23 +1127,23 @@ func NewSparkExecutor(execType pc.SparkExecutorType, config pc.SparkExecutorConf
 	case pc.EMR:
 		emrConfig, ok := config.(*pc.EMRConfig)
 		if !ok {
-			return nil, fmt.Errorf("cannot convert config into 'EMRConfig'")
+			return nil, fferr.NewInvalidArgument(fmt.Errorf("cannot convert config into 'EMRConfig'"))
 		}
 		return NewEMRExecutor(*emrConfig, logger)
 	case pc.Databricks:
 		databricksConfig, ok := config.(*pc.DatabricksConfig)
 		if !ok {
-			return nil, fmt.Errorf("cannot convert config into 'DatabricksConfig'")
+			return nil, fferr.NewInvalidArgument(fmt.Errorf("cannot convert config into 'DatabricksConfig'"))
 		}
 		return NewDatabricksExecutor(*databricksConfig)
 	case pc.SparkGeneric:
 		sparkGenericConfig, ok := config.(*pc.SparkGenericConfig)
 		if !ok {
-			return nil, fmt.Errorf("cannot convert config into 'SparkGenericConfig'")
+			return nil, fferr.NewInvalidArgument(fmt.Errorf("cannot convert config into 'SparkGenericConfig'"))
```

Comment:
Fee like this should be an internal error if we allow this struct and type to get passed through together